src/services/codefixes/fixUnreachableCode.ts

```diff
@@ -36,14 +36,33 @@ namespace ts.codefix {
                 break;
             default:
                 if (isBlock(statement.parent)) {
-                    split(sliceAfter(statement.parent.statements, statement), s => !isFunctionDeclaration(s), (start, end) => changes.deleteNodeRange(sourceFile, start, end));
+                    split(sliceAfter(statement.parent.statements, statement), shouldRemove, (start, end) => changes.deleteNodeRange(sourceFile, start, end));
                 }
                 else {
                     changes.deleteNode(sourceFile, statement);
                 }
         }
     }
 
+    function shouldRemove(s: Statement): boolean {
+        // Don't remove statements that can validly be used before they appear.
+        return !isFunctionDeclaration(s) && !isPurelyTypeDeclaration(s) &&
+            // `var x;` may declare a variable used above
+            !(isVariableStatement(s) && !(getCombinedNodeFlags(s) & (NodeFlags.Let | NodeFlags.Const)) && s.declarationList.declarations.some(d => !d.initializer));

```

Comment:
if it is a var statement it can not have an initializer, and it is not ambient.

Reply:
If it's ambient, it should not have an initializer so we won't remove it, right?